internal/agent/tools/bash.go

```diff
@@ -35,6 +35,115 @@ var BashToolDefinition = openai.Tool{
 	},
 }
 
+// GenerateCommandRegex generates a regex pattern from a bash command
+// The pattern is specific enough to match similar commands but general enough to be useful
+// For example:
+// - Command: "ls -la /tmp" → Regex: "^ls.*"
+// - Command: "git status" → Regex: "^git status.*"
+// - Command: "npm install package" → Regex: "^npm install.*"
+func GenerateCommandRegex(command string) string {
+	// Split the command into parts
+	parts := strings.Fields(command)
+
+	// If we have no parts, return a pattern that won't match anything
+	if len(parts) == 0 {
+		return "^$"
+	}
+
+	// For most commands, we'll use the first part as the base
+	// For certain commands like git, we might want to include the first two parts
+	baseCommand := parts[0]
+
+	// Special handling for commands that have meaningful subcommands
+	// like git, npm, etc.
+	specialCommands := map[string]bool{
```

Comment:
hmmm it's very hard for us to get this list right. There are way more programs out there that are worth being treated the same as these ones. it would be great to avoid hard coding the list and instead opt for a more generally applicable logic - thoughts?